src/python-common/ceph/deployment/service_spec.py

```diff
@@ -140,17 +140,111 @@ def validate(self) -> None:
         assert_valid_host(self.hostname)
 
 
+HostPatternType = Union[str, None, Dict[str, Union[str, bool, None]], "HostPattern"]
+
+
+class HostPattern():
+    def __init__(self, pattern: Optional[str] = None, is_regex: bool = False) -> None:
+        self.pattern = pattern
+        self.is_regex = is_regex
+
+    def filter_hosts(self, hosts: List[str]) -> List[str]:
+        if not self.pattern:
+            return []
+        if not self.is_regex:
+            return fnmatch.filter(hosts, self.pattern)
+        else:
+            return [h for h in hosts if re.match(self.pattern, h)]
```

Comment:
I think re.match is a good choice. I think it's better to use it here than re.search.